### ISSUE TITLE
`add_marker` not working while plot is not active

### ISSUE THREAD

**Issue by GeriTopore** [hyperspy/hyperspy#2799]
I am having this issue with the `add_marker()` method of `Signal1D` while working on Jupyter notebooks:
```python
import hyperspy.api as hs
import numpy as np

s = hs.signals.Signal1D(np.arange(500).reshape([5,10,10]))

marker_array = np.tile(np.arange(10), (5,1))
m = hs.plot.markers.vertical_line(x=marker_array)
s.add_marker(m)
```

The above code works perfectly fine if it is run in a single cell; the signal will be created and plotted with the markers in place. However if I split the code in different cells like below:
```python
# cell no. 1
import hyperspy.api as hs
import numpy as np

s = hs.signals.Signal1D(np.arange(500).reshape([5,10,10]))

#cell no. 2
marker_array = np.tile(np.arange(10), (5,1))
m = hs.plot.markers.vertical_line(x=marker_array)
s.add_marker(m)
```
and follow these steps:
1. Create the signal `s` by running cell no. 1.
2. Run cell no. 2 to create `marker_array` and marker object `m` and add to `s` using `add_marker()`.
3. Close the plot that appears.
4. Try to run cell no. 2 again.

I get this error:
```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-36-7302ff075bd5> in <module>
      1 marker_array = np.tile(np.arange(10), (5,1))
      2 m = hs.plot.markers.vertical_line(x=marker_array)
----> 3 s.add_marker(m)

c:\users\geri\documents\dev\hyperspy\hyperspy\signal.py in add_marker(self, marker, plot_on_signal, plot_marker, permanent, plot_signal, render_figure)
   5791                     self.plot()
   5792                 if m._plot_on_signal:
-> 5793                     self._plot.signal_plot.add_marker(m)
   5794                 else:
   5795                     if self._plot.navigator_plot is None:

AttributeError: 'NoneType' object has no attribute 'add_marker'
```

Now if I run `s.plot()` and then run cell no. 2 again this error disappears and the markers are put on the plot, but it will always give me this error if the plot isn't active.

Any ideas?

**Comment by ericpre** [comment]
Thanks @GeriTopore, I can reproduce it. The issue is most likely that when closing the plot, `signal_plot` and `navigation_plot` are set to `None` somewhere around https://github.com/hyperspy/hyperspy/blob/1eb0bdfaafa85b07750853141bde9921df479f30/hyperspy/drawing/mpl_he.py#L221-L230 and it meanst that checking if the plot is active needs to be fixed accordingly in https://github.com/hyperspy/hyperspy/blob/1eb0bdfaafa85b07750853141bde9921df479f30/hyperspy/signal.py#L5789-L5791

Using `self._plot.is_active` should fix it!

**Comment by ericpre** [comment]
Fixed in #2802.
